url29ff2e53f3_nicolas3470_Settlers_of_Catan_AI_tgz-pJ8-robotGameJ8/src/soc/robot/SOCRobotNegotiator.java

```diff
@@ -0,0 +1,2519 @@
+       + ourNewFavoriteCityETA + ourNewFavoriteRoadETA + ourNewPossibleCardETA));
+       D.ebugPrintln("-- ourOriginalFavoriteSettlementScore: "+ourOriginalFavoriteSettlementScore);
+       D.ebugPrintln("--       ourOriginalFavoriteCityScore: "+ourOriginalFavoriteCityScore);
+       D.ebugPrintln("--       ourOriginalFavoriteRoadScore: "+ourOriginalFavoriteRoadScore);
+       D.ebugPrintln("--       ourOriginalPossibleCardScore: "+ourOriginalPossibleCardScore);
+       D.ebugPrintln("--                            total: "+(ourOriginalFavoriteSettlementScore
+       + ourOriginalFavoriteCityScore + ourOriginalFavoriteRoadScore + ourOriginalPossibleCardScore));
+       D.ebugPrintln("-- ourNewFavoriteSettlementScore: "+ourNewFavoriteSettlementScore);
+       D.ebugPrintln("--       ourNewFavoriteCityScore: "+ourNewFavoriteCityScore);
+       D.ebugPrintln("--       ourNewFavoriteRoadScore: "+ourNewFavoriteRoadScore);
+       D.ebugPrintln("--       ourNewPossibleCardScore: "+ourNewPossibleCardScore);
+       D.ebugPrintln("--                            total: "+(ourNewFavoriteSettlementScore
+       + ourNewFavoriteCityScore + ourNewFavoriteRoadScore + ourNewPossibleCardScore));
+    
+       //
+       // see if we have something to gain from the offer
+       //
+       if (ourOriginalPieceType == ourNewPieceType) {
+       //
+       //  things to check if we want to build the
+       //  same piece before and after the deal
+       //
+       if ((ourOriginalPieceETA > ourNewPieceETA) ||
+       ((ourOriginalPieceETA == ourNewPieceETA) &&
+       (ourOriginalPieceScore < ourNewPieceScore))) {
+       response = ACCEPT_OFFER;
+       } else {
+       response = COUNTER_OFFER;
+       //brain.getClient().sendText(game, "That deal's not good for me.");
+       }
+       } else {
+       //
+       //  things to check if we changed our
+       //  plans based on the deal
+       //
+       if ((ourOriginalPieceScore < ourNewPieceScore) ||
+       ((ourOriginalPieceScore == ourNewPieceScore) &&
+       (ourOriginalPieceETA > ourNewPieceETA))) {
+       response = ACCEPT_OFFER;
+       } else {
+       response = COUNTER_OFFER;
+       //brain.getClient().sendText(game, "That deal's not good for me.");
+       }
+       }
+     */
+    /*
+       if (response == ACCEPT_OFFER) {
+       //
+       //  if the deal allows them to build a piece now,
+       //  see if their piece has a higher score than ours
+       //
+       if ((theirNewPieceETA == 0) &&
+       (theirNewPieceScore > ourNewPieceScore)) {
+       brain.getClient().sendText(game, "I think you're comming out ahead on that deal.");
+       response = REJECT_OFFER;
+       }
+       }
+     */
+    /*
+       }
+       }
+       }
+       return response;
+       }
+     */
+
+    /**
+     * @return a counter offer or null
+     *
+     * @param originalOffer  the offer given to us
+     */
+    public SOCTradeOffer makeCounterOffer(SOCTradeOffer originalOffer)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 123. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 1673-1687 into a separate method.